src/Testcontainers.Typesense/TypesenseBuilder.cs

```diff
@@ -0,0 +1,122 @@
+
+    public const ushort DefaultPort = 8108;
+
+    public const string DefaultApiKey = "iXy3xHM!KRogYBLg&nRg0V";
+
+    public const bool DefaultCors = true;
+
+    public const string DefaultVolume = "/data";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseBuilder" /> class.
+    /// </summary>
+    public TypesenseBuilder()
+        : this(new TypesenseConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TypesenseBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private TypesenseBuilder(TypesenseConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override TypesenseConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Sets the Api key.
+    /// </summary>
+    /// <param name="apiKey">The Api key.</param>
+    /// <returns>A configured instance of <see cref="TypesenseBuilder" />.</returns>
+    public TypesenseBuilder WithApiKey(string apiKey)
+    {
+        return Merge(DockerResourceConfiguration, new TypesenseConfiguration(apiKey: apiKey))
+          .WithCommand("--api-key", apiKey);
+    }
+
+    /// <summary>
+    /// Enables Cors.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="TypesenseBuilder" />.</returns>
+    public TypesenseBuilder EnableCors(bool enableCors)
+    {
+        if (enableCors)
+        {
+            return Merge(DockerResourceConfiguration, new TypesenseConfiguration(enableCors: enableCors))
+              .WithCommand("--enable-cors");
+        }
+
+        return Merge(DockerResourceConfiguration, new TypesenseConfiguration(enableCors: enableCors));
+    }
+
+    /// <summary>
+    /// Enables Cors.
+    /// </summary>
+    /// <returns>A configured instance of <see cref="TypesenseBuilder" />.</returns>
+    public TypesenseBuilder WithDataDirectory(string volume)
+    {
+        return Merge(DockerResourceConfiguration, new TypesenseConfiguration(volume: volume))
+          .WithCommand("--data-dir", volume);
+    }
+
+    /// <inheritdoc />
+    public override TypesenseContainer Build()
+    {
+        Validate();
+        return new TypesenseContainer(DockerResourceConfiguration);
+    }
+
+    /// <inheritdoc />
+    protected override TypesenseBuilder Init()
+    {
+        return base.Init()
+            .WithImage(TypesenseImage)
+            .WithPortBinding(DefaultPort, true)
+            .WithApiKey(DefaultApiKey)
+            .EnableCors(DefaultCors)
+            .WithTmpfsMount(DefaultVolume)
+            .WithDataDirectory(DefaultVolume)
+            .WithWaitStrategy(Wait.ForUnixContainer().UntilMessageIsLogged("Peer refresh succeeded"));
```

Comment:
Typesense offers a health endpoint. You can use the http strategy instead. See [java implementation](https://github.com/testcontainers/testcontainers-java/blob/main/modules/typesense/src/main/java/org/testcontainers/typesense/TypesenseContainer.java#L33-L37)